src/SFA.DAS.ApprenticeCommitments/Data/Models/ApprenticeCommitmentsDbContext.cs

```diff
@@ -105,5 +116,22 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             });
             base.OnModelCreating(modelBuilder);
         }
+
+        public override async Task<int> SaveChangesAsync(bool acceptAllChangesOnSuccess, CancellationToken cancellationToken = default)
+        {
+            var i = await base.SaveChangesAsync(acceptAllChangesOnSuccess, cancellationToken);
+            await DispatchDomainEvents();
+            return i;
+        }
```

Comment:
just curious as to why we are returning i, and what does it represent?

Reply:
The base `SaveChangesAsync` returns the number of rows that are updates.

Reply:
I have renamed it to `numEntriesWritten` to be clearer.